third_party/intellij-plugins-dart/src/com/jetbrains/lang/dart/ide/runner/server/vmService/VmOpenSourceLocationListener.java

```diff
@@ -0,0 +1,206 @@
+package com.jetbrains.lang.dart.ide.runner.server.vmService;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import de.roderick.weberknecht.WebSocket;
+import de.roderick.weberknecht.WebSocketEventHandler;
+import de.roderick.weberknecht.WebSocketException;
+import de.roderick.weberknecht.WebSocketMessage;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * TODO(cbernaschina) remove when changes to Dart Plugin land
+ *
+ * This class manages the openSourceLocation VM service registration and exposes the request as an event to listeners.
+ */
+public class VmOpenSourceLocationListener {
+
+  private interface MessageSender {
+    void sendMessage(JsonObject message);
+    void close();
+  }
+
+  public interface Listener {
+    void onRequest(@NotNull String isolateId, @NotNull String scriptId, int tokenPos);
+  }
+
+  /**
+   * Connect to the VM observatory service via the specified URI
+   *
+   * @return an API object for interacting with the VM service (not {@code null}).
+   */
+  public static VmOpenSourceLocationListener connect(@NotNull final String url) throws IOException {
+
+    // Validate URL
+    final URI uri;
+    try {
+      uri = new URI(url);
+    }
+    catch (URISyntaxException e) {
+      throw new IOException("Invalid URL: " + url, e);
+    }
+    String wsScheme = uri.getScheme();
+    if (!"ws".equals(wsScheme) && !"wss".equals(wsScheme)) {
+      throw new IOException("Unsupported URL scheme: " + wsScheme);
+    }
+
+    // Create web socket and observatory
+    final WebSocket webSocket;
+    try {
+      webSocket = new WebSocket(uri);
+    }
+    catch (WebSocketException e) {
+      throw new IOException("Failed to create websocket: " + url, e);
+    }
+    final VmOpenSourceLocationListener listener = new VmOpenSourceLocationListener(new MessageSender() {
+      @Override
+      public void sendMessage(JsonObject message) {
+        try {
+          webSocket.send(message.toString());
+        }
+        catch (WebSocketException e) {
+          // ignore
+          e.printStackTrace();
```

Comment:
Let's log this instead; there are some examples in the codebase of creating loggers based on the current class and logging to them (you can search for `LOG.`).